Spec/PushChannel.swift

```diff
@@ -0,0 +1,330 @@
+
+                    expect(query).to(haveParam("deviceId", withValue: params["deviceId"]))
+                    expect(query).toNot(haveParam("clientId", withValue: rest.device.clientId))
+                    expect(query).to(haveParam("channel", withValue: params["channel"]))
+                    expect(query).to(haveParam("concatFilters", withValue: "true"))
+                }
+
+                it("should return a paginated result with PushChannelSubscription filtered by channel and client") {
+                    let params = [
+                        "clientId": "tester",
+                        "channel": "aaa"
+                    ]
+                    let channel = rest.channels.get("foo")
+                    waitUntil(timeout: testTimeout) { done in
+                        try? channel.push.listSubscriptions(params) { result, error in
+                            expect(error).to(beNil())
+                            expect(result).toNot(beNil())
+                            done()
+                        }
+                    }
+
+                    guard let request = mockHttpExecutor.requests.first else {
+                        fail("should have a \"/push/channelSubscriptions\" request"); return
+                    }
+                    guard let url = request.url, url.absoluteString.contains("/push/channelSubscriptions") else {
+                        fail("should have a \"/push/channelSubscriptions\" URL"); return
+                    }
+                    guard let query = request.url?.query else {
+                        fail("should have a body"); return
+                    }
+
+                    expect(query).to(haveParam("clientId", withValue: params["clientId"]))
+                    expect(query).toNot(haveParam("deviceId", withValue: rest.device.id))
+                    expect(query).to(haveParam("channel", withValue: params["channel"]))
+                    expect(query).to(haveParam("concatFilters", withValue: "true"))
+                }
+
+                it("should not accept null deviceId and null clientId") {
+                    let channel = rest.channels.get("foo")
+                    expect { try channel.push.listSubscriptions([:]) { _, _ in } }.to(throwError { (error: NSError) in
+                        expect(error.code).to(equal(ARTDataQueryError.missingRequiredFields.rawValue))
+                    })
+                }
+
+                it("should not accept both deviceId and clientId params at the same time") {
+                    let params = [
+                        "deviceId": "x",
+                        "clientId": "y"
+                    ]
+                    let channel = rest.channels.get("foo")
+                    expect { try channel.push.listSubscriptions(params) { _, _ in } }.to(throwError { (error: NSError) in
+                        expect(error.code).to(equal(ARTDataQueryError.invalidParameters.rawValue))
+                    })
+                }
```

Comment:
Please add a test in which you actually subscribe and then you check that the result of `listSubscriptions` actually includes the result of the subscription.

Reply:
Done 720d602 but it doesn't pass. I already reported that issue in Slack.

Reply:
If you put a breakpoint on [this line](https://github.com/ably/ably-ios/commit/720d602766f712a8ace8bb4d7d75ac667fe4b2ea#diff-d254ef992ef12123539ed0708d7375d1R337) you'll get

```
(lldb) po channel.name
"test-2-pushenabled:foo"

(lldb) 
```

Reply:
@ricardopereira quite sure it's due to this https://github.com/ably/ably-ios/blob/push-channels/Spec/TestUtilities.swift#L100

Reply:
Fixed in https://github.com/ably/ably-ios/pull/708/commits/02b4d05740d7edf1340f051e1bc69853db8a79cb